services/vision/vision.go

```diff
@@ -127,8 +128,19 @@ func FindFirstName(r robot.Robot) string {
 
 // FirstFromRobot returns the first vision service in this robot.
 func FirstFromRobot(r robot.Robot) (Service, error) {
-	name := FindFirstName(r)
-	return FromRobot(r, name)
+	return FirstFromRobotMain(r)
+}
+
+// FirstFromRobotMain returns name of first vision service on this main robot.
+// This will specifically ignore remote resources.
+func FirstFromRobotMain(r robot.Robot) (Service, error) {
+	for _, n := range r.ResourceNames() {
+		if n.Subtype == Subtype && n.Remote == "" {
```

Comment:
drive by but `!n.ContainsRemoteNames()` would be more future-proof

Reply:
Love this drive by thank you!